test/scriptFlywheel/unit/borrowRatePerBlockTests.spec.ts

```diff
@@ -0,0 +1,183 @@
+        bankContract = await testEngine.bank;
+        // 1. initialization.
+        tokens = await testEngine.erc20Tokens;
+        mockChainlinkAggregators = await testEngine.mockChainlinkAggregators;
+        addressDAI = tokens[0];
+        addressUSDC = tokens[1];
+        addressMKR = tokens[4];
+        addressTUSD = tokens[3];
+        addressUSDT = tokens[2];
+
+        addressWBTC = tokens[8];
+
+        mockChainlinkAggregatorforTUSDAddress = mockChainlinkAggregators[3];
+        mockChainlinkAggregatorforDAIAddress = mockChainlinkAggregators[0];
+        mockChainlinkAggregatorforUSDCAddress = mockChainlinkAggregators[1];
+        mockChainlinkAggregatorforETHAddress = mockChainlinkAggregators[9];
+        erc20DAI = await ERC20.at(addressDAI);
+        erc20USDC = await ERC20.at(addressUSDC);
+        erc20MKR = await ERC20.at(addressMKR);
+        erc20TUSD = await ERC20.at(addressTUSD);
+        mockChainlinkAggregatorforDAI = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforDAIAddress
+        );
+        mockChainlinkAggregatorforUSDC = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforUSDCAddress
+        );
+        mockChainlinkAggregatorforETH = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforETHAddress
+        );
+        mockChainlinkAggregatorforTUSD = await MockChainLinkAggregator.at(
+            mockChainlinkAggregatorforTUSDAddress
+        );
+        cDAI_addr = await testEngine.tokenInfoRegistry.getCToken(addressDAI);
+        cUSDC_addr = await testEngine.tokenInfoRegistry.getCToken(addressUSDC);
+        cUSDT_addr = await testEngine.tokenInfoRegistry.getCToken(addressUSDT);
+        cWBTC_addr = await testEngine.tokenInfoRegistry.getCToken(addressWBTC);
+        cDAI = await MockCToken.at(cDAI_addr);
+        cUSDC = await MockCToken.at(cUSDC_addr);
+        cUSDT = await MockCToken.at(cUSDT_addr);
+        cWBTC = await MockCToken.at(cWBTC_addr);
+        numOfToken = new BN(1000);
+        ONE_DAI = eighteenPrecision;
+        ONE_ETH = eighteenPrecision;
+        ONE_USDC = sixPrecision;
+
+        await savingAccount.fastForward(1000);
+    });
+
+    context("getBorrowRatePerBlock()", async () => {
+        context("with 18 decimal Token", async () => {
+            context("Compound supported tokens", async () => {
+                context("should succeed", async () => {    
+                    it("when U = 1 for Compound supported tokens");
+    
+                    it("when U is between 0.9999 & 1 for Compound supported tokens");
+
+                    it("when U is between 0.90 and 0.5 for Compound supported tokens");
+
+                    it("when U is between 0.8999 & 0.90 for Compound supported tokens");
+
+                    it("when U is between 0.80 and 0.85 for Compound supported tokens");
+                });
+            });
+
+            context("Compound unsupported tokens", async () => {
```

Comment:
We dont have any test cases for different decimal tokens for compound unsupported tokens. Pls add similar tests like compound supported.

Reply:
I had previously added tests for those, they are present here: https://github.com/DeFinerOrg/Savings/blob/jitendra/non-comp-rate2/test/scriptFlywheel/unit/capUtilRatioTests.spec.ts

I think we should we merge these 2 files so all tests are in a single file.

Reply:
yes we can merge files